File: my_py_pkg/my_py_pkg/hw_status_publisher.py
import rclpy 
from rclpy.node import Node 

# import create custom msg
from my_robot_interfaces.msg import HardwareStatus

class HardwareStatusPublisherNode(Node): 
    def __init__(self): 
        super().__init__("hw_status_publisher") 
        self.publisher = self.create_publisher(HardwareStatus, "hardware_status", 10)
        self.timer = self.create_timer(0.5, self.publish_hw_status)
        self.get_logger().info("Hardware Status Publisher has been Started")

    def publish_hw_status(self): 
        msg = HardwareStatus() 
        msg.temperature = 45 
        msg.are_motors_ready = True 
        msg.debug_message = "this is custom msg"
        self.publisher.publish(msg)

def main(args=None): 
    rclpy.init(args=args)
    node = HardwareStatusPublisherNode() 
    rclpy.spin(node)
    rclpy.shutdown() 

if __name__ == "__main__": 
    main()

File: my_cpp_pkg/src/hw_status_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "my_robot_interfaces/msg/hardware_status.hpp" 

class HardwareStatusNode: public rclcpp::Node 
{ 
    public: 
        HardwareStatusNode(): Node("hw_status_publisher")
        { 
            RCLCPP_INFO(this->get_logger(), "Hardware Status Publisher has been started");  
            
            publisher = this->create_publisher<my_robot_interfaces::msg::HardwareStatus>("hardware_status", 10); 
            
            timer = this->create_wall_timer( 
                std::chrono::seconds(1), 
                std::bind(&HardwareStatusNode::PublishHwStatus, this)
            );

        } 

    private: 
        void PublishHwStatus()
        { 
            auto msg = my_robot_interfaces::msg::HardwareStatus(); 
            msg.temperatures = 45; 
            msg.are_motors_ready = true; 
            msg.debug_message = "this is custom msg"; 

            publisher->publish(msg);
            
        }
        
        rclcpp::Publisher<my_robot_interfaces::msg::HardwareStatus>::SharedPtr publisher;
        rclcpp::TimerBase::SharedPtr timer;

};

int main(int argc, char** argv) 
{ 
    rclcpp::init(argc, argv); 
    auto node = std::make_shared<HardwareStatusNode>();
    rclcpp::spin(node); 
    rclcpp::shutdown(); 
    return 0;
}